bundles/org.openhab.binding.dbquery/src/main/java/org/openhab/binding/dbquery/internal/DatabaseBridgeHandler.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * @author Joan Pujol - Initial contribution
+ */
+@NonNullByDefault
+public abstract class DatabaseBridgeHandler extends BaseBridgeHandler {
+    private static final long RETRY_CONNECTION_ATTEMPT_TIME_SECONDS = 60;
+    private final Logger logger = LoggerFactory.getLogger(DatabaseBridgeHandler.class);
+    private Database database = Database.EMPTY;
+    private @Nullable ScheduledFuture<?> retryConnectionAttemptFuture;
+
+    public DatabaseBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        logger.trace("initialize bridge {}", getThing().getUID());
+        initConfig();
+
+        database = createDatabase();
+
+        connectDatabase();
+    }
+
+    private void connectDatabase() {
+        logger.debug("connectDatabase {}", database);
+        var completable = database.connect();
+        updateStatus(ThingStatus.UNKNOWN);
+        completable.thenAccept(result -> {
+            if (result) {
+                logger.trace("Succesfully connected to database {}", getThing().getUID());
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                logger.trace("Connect to database {} failed", getThing().getUID());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                if (retryConnectionAttemptFuture == null) {
+                    scheduleRetryConnectionAttempt();
+                }
+            }
+        });
+    }
+
+    protected void scheduleRetryConnectionAttempt() {
+        logger.trace("Scheduled retry connection attempt every {}", RETRY_CONNECTION_ATTEMPT_TIME_SECONDS);
+        retryConnectionAttemptFuture = Executors.newSingleThreadScheduledExecutor().scheduleWithFixedDelay(
+                this::connectDatabase, RETRY_CONNECTION_ATTEMPT_TIME_SECONDS, RETRY_CONNECTION_ATTEMPT_TIME_SECONDS,
+                TimeUnit.SECONDS);
+    }
+
+    protected abstract void initConfig();
+
+    @Override
+    public void dispose() {
+        cancelRetryConnectionAttemptIfPresent();
+        disconnectDatabase();
+    }
+
+    protected void cancelRetryConnectionAttemptIfPresent() {
+        ScheduledFuture<?> currentFuture = retryConnectionAttemptFuture;
+        if (currentFuture != null)
+            currentFuture.cancel(true);
+    }
+
+    private void disconnectDatabase() {
+        var completable = database.disconnect();
+        updateStatus(ThingStatus.UNKNOWN);
+        completable.thenAccept(result -> {
+            Bridge bridge = getBridge();
+            if (result) {
+                logger.trace("Successfully disconnected to database {}", bridge != null ? bridge.getUID() : "null");
+                updateStatus(ThingStatus.OFFLINE);
+            } else {
+                logger.trace("Disconnect to database {} failed", bridge != null ? bridge.getUID() : "null");
+                updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.COMMUNICATION_ERROR);
```

Comment:
Same as above. Provide clear and helpful reasons why the error or device went offline if possible and dont log it when the status changes.

Reply:
The same justification as above, I think that connection-disconnection are very important messages, and a very low-level debug category is used.
In case of problems, I think that will be very useful.

Reply:
I understand your points, but keeping the logs clean from redundant messages is important. See https://next.openhab.org/docs/developer/guidelines.html#f-logging
```
void myFun() {
    logger.debug("And now the thing goes online"); // DONT, DONT, really DONT do that
    updateState(ThingState.ONLINE);
}
```

Reply:
I see your point and I will remove but I don't agree, I think that mainly all the persistence bindings had logs like this.
And the problem with the events.log is that if you do that way you can't have that relevant debug/trace logs only for an addon, can you?

For me is very useful to ask a user without a reproducible problem clear instructions, like set dbquery category log category to trace and send me the logs.

Reply:
That's right, the events.log contains the thing status detail messages of all Things.